### ISSUE TITLE
[Feature] FUSE Client

### ISSUE THREAD

**Issue by tobru** [eikek/docspell#717]
It would be very cool to be able to mount the Docspell instance as [FUSE](https://en.wikipedia.org/wiki/Filesystem_in_Userspace) mount to be able to browse the content of Docspell directly in the filesystem. It would be enough to be read-only, no need to be able to upload files.

One use-case is e.g. in CH when one has to do the  "Steuererklärung" a lot of documents have to be collected and finally be uploaded to the "Steueramt". Currently that means all the files in Docspell have to be manually downloaded and then uploaded again. With FUSE one could simply browse to the files like they are stored locally, without downloading first.

**Comment by eikek** [comment]
Yes, great idea! I'm in CH, too, and facing the same thing :-) There is something similar: #445. I thought about creating another process that creates a ftp-to-docspell or maybe ssh-to-docspel bridge. Then one would be able to use FTPFS, for example.

One little workaround for now might be tagging all the documents with a special tag and then using curl to download all with this tag into the local directory.

**Comment by eikek** [comment]
@tobru fyi, for the mentioned workaround, I just added a script in `tools/download-files.sh` for this. It was only a quick adaption to teh export-files script.

**Comment by JaCoB1123** [comment]
I extended my [docspell-cli](https://git.javil.eu/jacob1123/docspell-cli) to be able to create a FUSE mount. It also provides methods to convert path to url and vice versa. This allows me to copy docspell url, convert to path and then paste that in firefox to upload (or use e.g. ripdrag to get a draggable file) without downloading the file.
